# HG changeset 9714 patch
From: kaf24@firebug.cl.cam.ac.uk
# Node ID 32123a23a609440ee5c8ab7541f4c635f4b61dec
# Parent  db51d90a9bfefd5c43e315105fb5e1acfbaaf04a
Subject: Simplify shared_info mapping code. Always use a fixmap.
Signed-off-by: Keir Fraser <keir@xensource.com>
xen-unstable changeset:   10265:c9e6255cb44a2ec123b7bec9c9447697c747a5f3
xen-unstable date:        Fri Jun  2 19:14:44 2006 +0100

Acked-by: Jan Beulich <jbeulich@novell.com>

Index: head-2006-06-06/arch/i386/mm/init-xen.c
===================================================================
--- head-2006-06-06.orig/arch/i386/mm/init-xen.c	2006-06-06 12:51:13.000000000 +0200
+++ head-2006-06-06/arch/i386/mm/init-xen.c	2006-06-06 12:52:32.000000000 +0200
@@ -559,15 +559,11 @@ void __init paging_init(void)
 
 	kmap_init();
 
-	if (!xen_feature(XENFEAT_auto_translated_physmap) ||
-	    xen_start_info->shared_info >= xen_start_info->nr_pages) {
-		/* Switch to the real shared_info page, and clear the
-		 * dummy page. */
-		set_fixmap(FIX_SHARED_INFO, xen_start_info->shared_info);
-		HYPERVISOR_shared_info =
-			(shared_info_t *)fix_to_virt(FIX_SHARED_INFO);
-		memset(empty_zero_page, 0, sizeof(empty_zero_page));
-	}
+	/* Switch to the real shared_info page, and clear the
+	 * dummy page. */
+	set_fixmap(FIX_SHARED_INFO, xen_start_info->shared_info);
+	HYPERVISOR_shared_info = (shared_info_t *)fix_to_virt(FIX_SHARED_INFO);
+	memset(empty_zero_page, 0, sizeof(empty_zero_page));
 
 	/* Setup mapping of lower 1st MB */
 	for (i = 0; i < NR_FIX_ISAMAPS; i++)
Index: head-2006-06-06/arch/x86_64/kernel/setup-xen.c
===================================================================
--- head-2006-06-06.orig/arch/x86_64/kernel/setup-xen.c	2006-06-06 12:51:13.000000000 +0200
+++ head-2006-06-06/arch/x86_64/kernel/setup-xen.c	2006-06-06 12:52:32.000000000 +0200
@@ -670,13 +670,6 @@ void __init setup_arch(char **cmdline_p)
 
 	setup_xen_features();
 
-	if (xen_feature(XENFEAT_auto_translated_physmap) &&
-	    xen_start_info->shared_info < xen_start_info->nr_pages) {
-		HYPERVISOR_shared_info =
-			(shared_info_t *)__va(xen_start_info->shared_info);
-		memset(empty_zero_page, 0, sizeof(empty_zero_page));
-	}
-
 	HYPERVISOR_vm_assist(VMASST_CMD_enable,
 			     VMASST_TYPE_writable_pagetables);
 
Index: head-2006-06-06/arch/x86_64/mm/init-xen.c
===================================================================
--- head-2006-06-06.orig/arch/x86_64/mm/init-xen.c	2006-06-06 12:52:30.000000000 +0200
+++ head-2006-06-06/arch/x86_64/mm/init-xen.c	2006-06-06 12:52:32.000000000 +0200
@@ -757,15 +757,11 @@ void __init paging_init(void)
 	free_area_init_node(0, NODE_DATA(0), zones,
 			    __pa(PAGE_OFFSET) >> PAGE_SHIFT, holes);
 
-	if (!xen_feature(XENFEAT_auto_translated_physmap) ||
-	    xen_start_info->shared_info >= xen_start_info->nr_pages) {
-		/* Switch to the real shared_info page, and clear the
-		 * dummy page. */
-		set_fixmap(FIX_SHARED_INFO, xen_start_info->shared_info);
-		HYPERVISOR_shared_info =
-			(shared_info_t *)fix_to_virt(FIX_SHARED_INFO);
-		memset(empty_zero_page, 0, sizeof(empty_zero_page));
-	}
+	/* Switch to the real shared_info page, and clear the
+	 * dummy page. */
+	set_fixmap(FIX_SHARED_INFO, xen_start_info->shared_info);
+	HYPERVISOR_shared_info = (shared_info_t *)fix_to_virt(FIX_SHARED_INFO);
+	memset(empty_zero_page, 0, sizeof(empty_zero_page));
 
 	init_mm.context.pinned = 1;
 
Index: head-2006-06-06/include/asm-i386/mach-xen/setup_arch_post.h
===================================================================
--- head-2006-06-06.orig/include/asm-i386/mach-xen/setup_arch_post.h	2006-06-06 12:51:13.000000000 +0200
+++ head-2006-06-06/include/asm-i386/mach-xen/setup_arch_post.h	2006-06-06 12:52:32.000000000 +0200
@@ -25,13 +25,6 @@ static void __init machine_specific_arch
 	struct xen_platform_parameters pp;
 	struct xennmi_callback cb;
 
-	if (xen_feature(XENFEAT_auto_translated_physmap) &&
-	    xen_start_info->shared_info < xen_start_info->nr_pages) {
-		HYPERVISOR_shared_info =
-			(shared_info_t *)__va(xen_start_info->shared_info);
-		memset(empty_zero_page, 0, sizeof(empty_zero_page));
-	}
-
 	HYPERVISOR_set_callbacks(
 	    __KERNEL_CS, (unsigned long)hypervisor_callback,
 	    __KERNEL_CS, (unsigned long)failsafe_callback);
